src/core/manager.c

```diff
@@ -2937,8 +2937,10 @@ int manager_loop(Manager *m) {
                 usec_t wait_usec, watchdog_usec;
 
                 watchdog_usec = manager_get_watchdog(m, WATCHDOG_RUNTIME);
-                if (timestamp_is_set(watchdog_usec))
+                if (m->hw_watchdog_running)
```

Comment:
We have multiple types of watchdogs. What happens if another type is set and then we end up here? Eg: if the reboot watchdog is set, but the runtime is not, manager_set_watchdog will have set this bool to true, since it's unconditional and doesn't depend on the type.

Reply:
I could definitely be wrong, so please correct me if I am, but my understanding of the code is that because WATCHDOG_RUNTIME is specified in manager_get_watchdog(m, WATCHDOG_RUNTIME) here, manager_loop() only cares about the runtime watchdog and not the other two.  The hw_watchdog_running boolean I added is supposed to only be set true when the runtime watchdog is running, but I did notice where I set those values in the wrong places, so I fixed that, thanks!  I also renamed the variable and the function I added to clarify that they only deal with the runtime watchdog and not the other two.

Reply:
Yes now it's fine that the bool is set only in the runtime case, but before watchdog_ping() could have been called if one of the others was set. Now it's fine. Thanks!